ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/place.hpp

```diff
@@ -47,18 +48,26 @@ namespace ngraph
             bool is_output() const override;
             bool is_equal(Ptr another) const override { return this == another.get(); }
 
+            const InputModelPDPD& get_input_model() const { return m_input_model; }
+
             std::vector<std::string> get_names() const override { return m_names; }
+            void add_name(const std::string& name) { m_names.push_back(name); }
+            void set_name(const std::string& name) { m_names.assign({name}); }
+            void remove_name(const std::string& name)
+            {
+                m_names.erase(std::remove(m_names.begin(), m_names.end(), name), m_names.end());
+            }
 
-        private:
-            const InputModel& m_input_model;
+        protected:
+            const InputModelPDPD& m_input_model;
```

Comment:
By the way, why is it 'const reference' and not something like 'weak_ptr'?
I'm thinking about situations when model is destroyed by client, but Place object still exists. Like
```
Place::Ptr get_place_only() {
    InputModel::Ptr model = frontend.load("abc.pdmodel");
    return model.get_place_by_tensor_name("tensor");
}

auto place = get_place_only();
std::cout << place->is_input();
```
From API perspective, it is valid code and should not lead to crash. There can be an exception that model is destroyed, but not crash

Reply:
current pipeline: ctor InputModel -> ctor InputModelImpl (..., const InputModel&) -> LoadPlaces() -> Place(..., const InputModel&)
as I understood, you suggest
ctor InputModel -> ctor InputModelImpl (..., weak_ptr<InputModel>) -> LoadPlaces() -> Place(..., weak_ptr<InputModel>)

In this case, we need to inherit  InputModel from enable_shared_from_this and use shared_from_this in ctor
Unfortunately it's impossible to use shared_from_this() in ctors.